src/PHPStan/UnionToIntersectionTypeResolver.php

```diff
@@ -41,18 +41,7 @@ public function resolve(TypeNode $typeNode, NameScope $nameScope): ?Type
 		}
 
 		$type = $this->typeNodeResolver->resolve($arguments[0], $nameScope);
-		if ($type instanceof UnionType) {
-			return new IntersectionType($type->getTypes());
-		}
-
-        if ($type instanceof ObjectType) {
-            return $type;
-        }
-
-		if ($type instanceof TemplateType) {
-			return new UnionToIntersectionType($type);
-		}
 
-		return null;
+		return new UnionToIntersectionType($type);
```

Comment:
The code already worked without these changes. This is just simplification of the extension code.